.github/workflows/test.yml

```diff
@@ -1,19 +1,34 @@
 name: aiosmtpd CI
 
 on:
+  # This is for direct commit to master
   push:
     branches: [ master ]
+  # This is for PRs
   pull_request:
     branches: [ master ]
+  # Manual/on-demand
   workflow_dispatch:
+  release:
+    types: [ created, edited, published, prereleased, released ]
 
 jobs:
   test:
-    runs-on: ubuntu-latest
-    continue-on-error: true
     strategy:
+      # If a matrix fail, do NOT stop other matrix, let them run to completion
+      fail-fast: false
```

Comment:
When one matrix combination fails, do NOT cancel other matrix combinations. Let them all run to completion.